src/App.tsx

```diff
@@ -1,14 +1,69 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useState, useEffect, useCallback } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
-
+import { getTodos } from './api';
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState('');
+  const [selectedFilter, setSelectedFilter] = useState('all');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [loader, setLoader] = useState(true);
+  const [modalOpened, setModalOpened] = useState(false);
+
+  const getfilteredTodos = useCallback(() => {
+    const normQuery = query.toLowerCase().trim();
+    let filteredTodos = [...todos];
+
+    if (normQuery) {
+      filteredTodos = filteredTodos
+        .filter(todo => todo.title.toLowerCase().includes(normQuery));
+    }
+
+    switch (selectedFilter) {
+      case 'active':
+        return filteredTodos.filter(todo => !todo.completed);
+
+      case 'completed':
+        return filteredTodos.filter(todo => todo.completed);
+
+      default:
+        return filteredTodos;
+    }
+  }, [query, selectedFilter, todos]);
+
+  const visibleTodos = getfilteredTodos();
```

Comment:
It's better to use useMemo here, because it's complex calculation and we want to prevent recalculation